Lps 200225 lps 200227 lps 200228

lfbesada opened the pull request:
:kissing:

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

sandrodw3 commented:
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

lfbesada committed:
LPS-200227 Throws a LockedLayoutException if a user tries to change the classNameId or the classTypeId of a locked page

lfbesada committed:
LPS-200227 SF use the LayoutLockManager method

lfbesada committed:
LPS-200227 Adds new getLock signature to specify the layout and the user ID instead of get them from theme display

lfbesada committed:
LPS-200227 Prevents NPE if layout is null

lfbesada committed:
LPS-200227 Throws an UnsupportedOperationException if trying to change the content type and the feature flag is disabled

lfbesada committed:
LPS-200225 Adds back_url and back_url_title parameters

lfbesada committed:
LPS-200225 SF pass changeContentTypeURL as dropdownItem data instead of the additional properties

lfbesada committed:
LPS-200228 Adds a ModelListener for LayoutPageTemplateEntry to prevent the content type changes when the display page is used by at least one asset.

I am using a model listener for this because trying to check it in the service layer will cause a circular dependency between AssetDisplayPageEntryLocalService and LayoutPageEntryLocalService. Checking it only in the action command would leave open the possibility to change the content type of a used display page by invoking directly the service method.

lfbesada committed:
LPS-200225 SF remove redundant condition as it is going to be always true

lfbesada committed:
LPS-200225 Hide all the behavior under the feature flag

lfbesada committed:
LPS-200225 SF cache in a variable to prevent calculating for each display page template

lfbesada committed:
LPS-200225 Pass existsMappedContentType to DisplayPageVerticalCard and use it to disable the name link to the edition when the mapped content-type doesn't exist

lfbesada committed:
LPS-200225 Adds a method to determine if the display page's mapped content type exists

lfbesada committed:
LPS-200225 Improve performance and simplify by caching the type and subtype IDs and using the map to find the type/subtype for each display page template

lfbesada committed:
LPS-200225 Pass existsMappedContentType to DisplayPageActionDropdownItemsProvider and use it to prevent access directly to the edit action if the display page is mapped to a non-existing type or a subtype.

To do it, on edit action click, we are displaying the change content type popup, and redirect to the edition once the content-type is fixed.

So we are keeping the Edit action label but setting as dropdownItem action the changeContentType, In addition, at this point, we are setting as redirect in changeContentTypeURL the URL to edit the display page.

lfbesada committed:
LPS-200227 Handles LockedLayoutException if occurs performing the action

lfbesada committed:
LPS-200227 Take the draft layout lock before changing the content-type and release it after performing

lfbesada committed:
LPS-200228 Handles ModelListenerException if the original cause is RequiredLayoutPageTemplateEntryException. We are handling this in the action command instead of the LayoutPageTemplateEntryExceptionRequestHandlerUtil since this error only should be processed in that way for this action command.

lfbesada committed:
LPS-200228 Adds other error properties needed by the front side to display the error dialog

lfbesada committed:
LPS-200228 Extract usageCount to reuse

lfbesada committed:
LPS-200228 SF use RenderURLBuilder

lfbesada committed:
LPS-200228 Pass the hasUsages property to the front side and, in case hasUsages, pass the viewUsagesURL instead of the changeContentTypeURL because we will display an informative dialog instead the change type form

lfbesada committed:
LPS-200228 Pass the usage count and use it in building the edit action dropdownItem.

Is a very unusual case that a DPT has some use but the type/subtype doesn't exist. I can only think of either bad data or the implementation of the type/subtype not being available at that time. That is why in this case, given that we cannot allow changing the type, we will simply disable the edition

lfbesada committed:
LPS-200228 SF cache the URLs to prevent calculating twice

lfbesada committed:
TO DELETE --> testing of LPS-200227 and LPS-200228

lfbesada committed:
LPS-200225 Take into account that 0 is a valid classTypeId since it is the infoItemFormVariationKey of the Basic Documents

lfbesada committed:
LPS-200227 Baseline
